From 46a4d7a378372a028522cfba94e77a11c478b36c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Daniel=20Vr=C3=A1til?= <dvratil@kde.org>
Date: Fri, 4 May 2018 00:31:44 +0200
Subject: Fix build against new Syndication library

Syndication::Loader::loadUrl() now always requires a DataRetriever
to be passed as a second argument. The behavior of the old method
was basically identical to using FeedRetriever anyway, so we can
use the FeedRetriever here to make it compile against new Syndication
and keep the behavior.
---
 plugins/syndication/ktfeed.cpp              | 2 +-
 plugins/syndication/syndicationactivity.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git ./plugins/syndication/ktfeed.cpp ./plugins/syndication/ktfeed.cpp
index 099cd4f..f20eb2c 100644
--- ./plugins/syndication/ktfeed.cpp
+++ ./plugins/syndication/ktfeed.cpp
@@ -277,7 +277,7 @@ namespace kt
         status = DOWNLOADING;
         update_timer.stop();
         Syndication::Loader* loader = Syndication::Loader::create(this, SLOT(loadingFromDiskComplete(Syndication::Loader*, Syndication::FeedPtr, Syndication::ErrorCode)));
-        loader->loadFrom(QUrl(dir + QStringLiteral("feed.xml")));
+        loader->loadFrom(QUrl(dir + QStringLiteral("feed.xml")), new FeedRetriever());
         updated();
     }
 
diff --git ./plugins/syndication/syndicationactivity.cpp ./plugins/syndication/syndicationactivity.cpp
index 293540d..61e8039 100644
--- ./plugins/syndication/syndicationactivity.cpp
+++ ./plugins/syndication/syndicationactivity.cpp
@@ -135,7 +135,7 @@ namespace kt
         }
         else
         {
-            loader->loadFrom(QUrl(url));
+            loader->loadFrom(QUrl(url), new FeedRetriever());
             downloads.insert(loader, url);
         }
     }
-- 
cgit v0.11.2

